go/saltpack/sign_stream.go

```diff
@@ -130,14 +132,33 @@ func newSignDetachedStream(w io.Writer, signer SigningSecretKey) (*signDetachedS
 		return nil, err
 	}
 
+	// Encode the header bytes.
+	headerBytes, err := encodeToBytes(header)
+	if err != nil {
+		return nil, err
+	}
+
+	// Compute the header hash.
+	headerDigest := sha512.New()
+	headerDigest.Write(headerBytes)
+	headerHash := headerDigest.Sum(nil)
```

Comment:
Let's factor this 3-line block out into a little helper function. Also let's look for other places this might be a thing.